deepes/test/src/torch_agent_test.cc

```diff
@@ -0,0 +1,153 @@
+//   Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gtest/gtest.h"
+
+#include <torch/torch.h>
+#include <memory>
+#include <algorithm>
+#include <glog/logging.h>
+#include <omp.h>
+#include "gaussian_sampling.h"
+#include "torch_demo_model.h"
+#include "es_agent.h"
+
+#include <random>
+#include <math.h>
+
+namespace DeepES {
+
+
+
+// The fixture for testing class Foo.
+class TorchDemoTest : public ::testing::Test {
+protected:
+    float evaluate(float* x_list, float* y_list, int size, std::shared_ptr<ESAgent<Model>> agent) {
+        float total_loss = 0.0;
+        for (int i = 0; i < size; ++i) {
+            torch::Tensor x_input = torch::tensor(x_list[i], torch::dtype(torch::kFloat32));
+            torch::Tensor predict_y = agent->predict(x_input);
+            auto pred_y = predict_y.accessor<float,2>();
+            float loss = pow((pred_y[0][0] - y_list[i]), 2);
+            total_loss += loss;
+        }
+        return -total_loss / float(size);
+    }
+
+    float train_loss() {
+        return -1.0 * evaluate(x_list, y_list, train_data_size, agent);
+    }
+
+    float test_loss() {
+        return -1.0 * evaluate(test_x_list, test_y_list, test_data_size, agent);
+    }
+
+    float train_test_gap() {
+        float train_lo = train_loss();
+        float test_lo = test_loss();
+        if ( train_lo > test_lo) {
+            return train_lo - test_lo;
+        }
+        else {
+            return test_lo - train_lo;
+        }
+    }
+
+    float train_agent() {
+    }
+
+
+    void SetUp() override {
+        x_list = new float [train_data_size];
```

Comment:
Please use vector<float> for data storage. It's more readable.

Reply:
done